src/components/Room/Room.js

```diff
@@ -496,42 +487,47 @@ function Room() {
    * @function receiveJoinRoomResponse - response after requesting to join a Room
    * @param {Object} response 
    */
-   function receiveJoinRoomResponse(response) {
+   function receiveJoinRoomResponse(joinRoomRequest, response) {
     let roomId = response.getRoomId();
     let status = response.getStatus();
 
-    if (status === 'accept') {
+    if (status === 'accept' || status == 'wait') {
+      let wait = status == 'wait';
+      if (wait) {
+        dispatch(setIsStrict(true));
+      }
+      // join the vestibule on any "positive" response from the RAS
       let vestibulePayload = {
         roomId: roomId,
-        roomPassword: null,
-        userName: null,
+        roomPassword: joinRoomRequest['roomPassword'],
+        roomUserName: joinRoomRequest['userName'],
       };
       dispatch(setVestibuleJoin(vestibulePayload));
 
-      // cleanup chatHistory json
-      let chatHistory = response.getChatHistoryList();
-
-      let chatHistoryData = [];
-      for (let i = 0; i < chatHistory.length; i++) {
-        chatHistoryData.push({
-          userId: chatHistory[i].getUserId(),
-          name: chatHistory[i].getUserName(),
-          message: chatHistory[i].getMessage(),
-          timestamp: chatHistory[i].getTimestamp(),
-        });
+      // Upon acceptance, capture and set additional state
+      if (!wait) {
+        console.log('r.receiveJoinRoomResponse: Accepted into roomId=%o', roomId);
+        let token = response.getToken();
+        let chatHistory = response.getChatHistoryList();
+        let chatHistoryData = [];
+
+        for (let i = 0; i < chatHistory.length; i++) {
+          chatHistoryData.push({
+            userId: chatHistory[i].getUserId(),
+            name: chatHistory[i].getUserName(),
+            message: chatHistory[i].getMessage(),
+            timestamp: chatHistory[i].getTimestamp(),
+          });
+        }
+        dispatch(setToken(token));
+        dispatch(setRoomJoinRequestAccepted(true));
+        dispatch(setChatHistory(chatHistoryData));
+        console.log('r.receiveJoinRoomResponse: userId=%o is ready to enter roomId=%o', store.getState().user.userId, roomId)
```

Comment:
might be minor typo in the loggers with `r.` or just ignore b/c these don't really matter and can be deleted.

Reply:
oh I thought I saw you do this elsewhere. I'll remove.